lib/commands/certificate.js

```diff
@@ -0,0 +1,166 @@
+import _ from 'lodash';
+import { fs, plist } from 'appium-support';
+import x509 from 'x509';
+import UUID from 'uuid-js';
+import path from 'path';
+import B from 'bluebird';
+import { waitForCondition } from 'asyncbox';
+import { openUrl } from 'node-simctl';
+
+
+let extensions = {}, commands = {};
+
+const SELF_MODULE_NAME = 'appium-xcuitest-driver';
+const BASE_MODULE_NAME = 'appium-base-driver';
+const CONFIG_EXTENSION = '.mobileconfig';
+
+
+async function toMobileConfig (cert) {
+  let altNames = [];
+  try {
+    altNames = x509.getAltNames(cert);
+  } catch (err) {
+    throw new Error(`Cannot parse the certificate. Is it valid and base64-encoded? ` +
+                    `Original error: ${err.message}`);
+  }
+  const displayName = _.replace(_.last(altNames), /\*/g, '_');
+  if (_.isEmpty(displayName)) {
+    throw new Error(`Cannot parse alternate names from the certificate. ` +
+                    `Original value: ${JSON.stringify(altNames)}`);
+  }
+  const uuid = () => UUID.create().hex.toUpperCase();
+  const contentUuid = uuid();
+  return {
+    PayloadContent: [{
+      PayloadCertificateFileName: `${displayName}.cer`,
+      PayloadContent: cert,
+      PayloadDescription: 'Adds a CA root certificate',
+      PayloadDisplayName: displayName,
+      PayloadIdentifier: `com.apple.security.root.${contentUuid}`,
+      PayloadType: 'com.apple.security.root',
+      PayloadUUID: contentUuid,
+      PayloadVersion: 1
+    }],
+    PayloadDisplayName: displayName,
+    PayloadIdentifier: `localhost.${uuid()}`,
+    PayloadRemovalDisallowed: false,
+    PayloadType: 'Configuration',
+    PayloadUUID: uuid(),
+    PayloadVersion: 1
+  };
+}
+
+async function clickElement (driver, locator, timeout = 5000) {
+  let element = null;
+  try {
+    await waitForCondition(async () => {
+      try {
+        element = await driver.findNativeElementOrElements(
+          locator.type, locator.value, false);
+        return true;
+      } catch (ign) {
+        return false;
+      }
+    }, {
+      waitMs: timeout,
+      intervalMs: 500,
+    });
+  } catch (err) {
+    throw new Error(`Cannot find ${JSON.stringify(locator)} within ${timeout}ms timeout`);
+  }
+  await driver.nativeClick(element);
+}
+
+async function installCertificateInPreferences (driver) {
+  // Accept Safari alert
+  await clickElement(driver, {
+    type: 'accessibility id',
+    value: 'Allow'
+  });
+
+  await B.delay(2000);
```

Comment:
shouldn't the `waitForCondition` in `clickElement` make it unnecessary to have a static wait here?